examples/material-ui-remix-ts/app/entry.server.tsx

```diff
@@ -1,59 +1,143 @@
-import * as React from 'react';
-import * as ReactDOMServer from 'react-dom/server';
-import { RemixServer } from '@remix-run/react';
-import type { EntryContext } from '@remix-run/node';
-import createEmotionCache from './src/createEmotionCache';
-import theme from './src/theme';
-import CssBaseline from '@mui/material/CssBaseline';
-import { ThemeProvider } from '@mui/material/styles';
-import { CacheProvider } from '@emotion/react';
-import createEmotionServer from '@emotion/server/create-instance';
+/**
+ * By default, Remix will handle generating the HTTP Response for you.
+ * You are free to delete this file if you'd like to, but if you ever want it revealed again, you can run `npx remix reveal` ✨
+ * For more information, see https://remix.run/file-conventions/entry.server
+ */
+
+import { PassThrough } from "node:stream";
+
+import type { AppLoadContext, EntryContext } from "@remix-run/node";
+import { createReadableStreamFromReadable } from "@remix-run/node";
+import { RemixServer } from "@remix-run/react";
+import { isbot } from "isbot";
+import { renderToPipeableStream } from "react-dom/server";
+import { MuiProvider } from "./mui/MuiProvider";
+
+const ABORT_DELAY = 5_000;
 
 export default function handleRequest(
   request: Request,
   responseStatusCode: number,
   responseHeaders: Headers,
   remixContext: EntryContext,
+  // This is ignored so we can keep it in the template for visibility.  Feel
+  // free to delete this parameter in your app if you're not using it!
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  loadContext: AppLoadContext
 ) {
-  const cache = createEmotionCache();
-  const { extractCriticalToChunks } = createEmotionServer(cache);
-
-  function MuiRemixServer() {
-    return (
-      <CacheProvider value={cache}>
-        <ThemeProvider theme={theme}>
-          {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
-          <CssBaseline />
-          <RemixServer context={remixContext} url={request.url} />
-        </ThemeProvider>
-      </CacheProvider>
-    );
-  }
+  return isbot(request.headers.get("user-agent") || "")
+    ? handleBotRequest(
```

Comment:
Why are you returning two different components in case is a bot or not?

Reply:
It's same as above. The issue is that you see lots of changes becuase it is compared to the old remix/mui example. I cloned a fresh copy of remix default template and applied MUI changes to it which is more straighforward. 

in order to get the right vibe from my changes, I encourage you to [see the changes](https://github.com/mahmoudmoravej/remix-mui/pull/1/files) I have done over the main Remix template files. For instance, you will see that the only chnages I have in `entry.server.tsx` is wrapping the `RemixServer` with the new `<MuiProvider>`